### PR TITLE
test: improve custom networking integration tests

### PR DESCRIPTION
**Pull request by yash97**
with cross-node enforcement , ClusterIP connectivity, and ASG-driven instance recycling

**What type of PR is this?**

**Which issue does this PR fix?**:

**What does this PR do / Why do we need it?**:

**Testing done on this change**:

**Will this PR introduce any new dependencies?**:

**Will this break upgrades or downgrades? Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

### New sample additional logs

```
  [DIAG] target ENI=eni-02ab8b6684e687eff subnet=subnet-03362a1f80ee1efc8 SGs=[sg-045787352f03c82a6]
```
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/integration/custom-networking/custom_networking_test.go [hunk 1]
```diff
@@ -90,7 +92,29 @@ var _ = Describe("Custom Networking Test", func() {
...
+				// Job can still schedule on 2-node clusters when targets span both nodes.
+				testJob.Spec.Template.Spec.Affinity = &coreV1.Affinity{
+					NodeAffinity: &coreV1.NodeAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: &coreV1.NodeSelector{
+							NodeSelectorTerms: []coreV1.NodeSelectorTerm{{
+								MatchExpressions: []coreV1.NodeSelectorRequirement{{
+									Key:      "kubernetes.io/hostname",
+									Operator: coreV1.NodeSelectorOpNotIn,
+									Values:   []string{pod.Spec.NodeName},
+								}},
```
[the remark this answers is not in the document]
Comment (yash97) [reply test/integration/custom-networking/custom_networking_test.go:108]: this is fine, we run integration test in controlled env.

File: test/framework/resources/aws/utils/nodegroup.go [hunk 2]
```diff
@@ -366,18 +369,64 @@ func TerminateInstances(f *framework.Framework) error {
...
+		p := metav1.DeletePropagationBackground
+		return &p
+	}()}
+	for i := range nodeList.Items {
+		_ = f.K8sResourceManagers.NodeManager().DeleteNode(&nodeList.Items[i], opts)
+	}
+
+	if err := f.CloudServices.AutoScaling().SetDesiredCapacity(context.TODO(), asgName, expected); err != nil {
+		return fmt.Errorf("failed to set desired capacity back to %d on %s: %v", expected, asgName, err)
+	}
```
[the remark this answers is not in the document]
Comment (yash97) [reply test/framework/resources/aws/utils/nodegroup.go:416]: We do notcreate cluster with multiple asg for particular label.

File: test/integration/custom-networking/custom_networking_test.go [hunk 3]
```diff
@@ -90,7 +92,29 @@ var _ = Describe("Custom Networking Test", func() {
...
+							}},
+						},
+					},
+				}
+
 				_, err := f.K8sResourceManagers.JobManager().CreateAndWaitTillJobCompleted(testJob)
+				logJobPodDiag(testJob.Name)
+				logTargetENIDiag(pod)
+
```
[the remark this answers is not in the document]
Comment (yash97) [reply test/integration/custom-networking/custom_networking_test.go:117]: This provides better understanding of how test was executed. We run this test at lower resource scale. Did not observed any throttling.

**Comment by haouc** [comment]
Seems need this small PR to cover edge case:
https://github.com/aws/amazon-vpc-cni-k8s/pull/3671

**Review by haouc — APPROVED** [review]
tested and looks good to me.
